packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -539,18 +552,18 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     // If there is at least one remaining block, we mine one. This way, we
     // guarantee that there's an empty block immediately before and after the
     // reservation. This makes the logging easier to get right.
-    if (count.gtn(mineBlockResults.length)) {
+    if (count > mineBlockResults.length) {
       await mineBlock();
     }
 
-    const remainingBlockCount = count.subn(mineBlockResults.length);
+    const remainingBlockCount = count - BigInt(mineBlockResults.length);
 
     // There should be at least 2 blocks left for the reservation to work,
     // because we always mine a block after it. But here we use a bigger
     // number to err on the safer side.
-    if (remainingBlockCount.lten(5)) {
+    if (remainingBlockCount <= 5) {
```

Comment:
wait, shouldn't this be `5n`?

Reply:
You can use mix numbers and bigints in comparison operators. But we should probably avoid doing that. Maybe there's an eslint rule for that.